app/repository/base_repository.py

```
@@ -0,0 +1,102 @@
+                raise NotFoundError(detail=f"not found id : {id}")
+            return query
+
+    def create(self, schema: T):
+        with self.session_factory() as session:
+            query = self.model(**schema.dict())
+            try:
+                session.add(query)
+                session.commit()
+                session.refresh(query)
+            except IntegrityError as e:
+                raise DuplicatedError(detail=str(e.orig))
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)